overlord/devicestate/handlers_install.go

```diff
@@ -651,6 +664,7 @@ func (m *DeviceManager) doFactoryResetRunSystem(t *state.Task, _ *tomb.Tomb) err
 		UnpackedGadgetDir: gadgetDir,
 
 		RecoverySystemLabel: modeEnv.RecoverySystem,
+		KernelMods:          bootKMods,
```

Comment:
no test seems to fail in the package if this is not passed in

Reply:
Test added